woocommerce/woocommerce-ios
# Add basic logic for add package action to get package data

Part of #13551


## Description


- Created **WooAddCustomPackageView** and put it into a separate file
- Added basic logic using **addPackageAction** closure to pass up the package data

Next step is to do a repository or some other model that would handle packages data and manipulation of them (getting them, creating, modifications, backend integration etc.)

## Testing information


1. Install and set up the Woo Shipping extension on your store.
2. Build and run the app with the `revampedShippingLabelCreation` feature flag enabled.
3. Create an order with the processing status and at least one physical product.
4. In the order details, select "Create Shipping Label."
5. Tap "Select a Package" button
6. Pick a package type, enter all the custom fields and tap "Add Package". The screen should be dismissed
7. In WooShippingPackageAndRatePlaceholder there is a comment "// TODO: use packageData", there you can put a breakpoint and some print/log to check the data from the closure
8. Do the same with Carrier and Saved tabs. There you just need to tap the package to select it and tap "Add Package"

## Screenshots


| Custom      | Carrier | Saved |
| ----------- | ----------- | ----------- |
| ![Simulator Screen Recording - iPhone 16 Pro - 2024-10-30 at 12 40 18](https://github.com/user-attachments/assets/5a38442a-463d-42bc-8d5b-95e3a6829676)      | ![Simulator Screen Recording - iPhone 16 Pro - 2024-10-30 at 12 40 32](https://github.com/user-attachments/assets/e38dcf37-e312-4101-9aa0-1d3c430633ca)       | ![Simulator Screen Recording - iPhone 16 Pro - 2024-10-30 at 12 40 43](https://github.com/user-attachments/assets/81b0a866-508b-46fa-aa9b-cd1caa180e18)       |

---


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
